File: sample/src/main/java/de/timfreiheit/mozart/sample/player/MusicService.java
package de.timfreiheit.mozart.sample.player;

import android.os.RemoteException;

import de.timfreiheit.mozart.MozartMediaNotificationManager;
import de.timfreiheit.mozart.MozartMusicService;
import de.timfreiheit.mozart.model.image.MozartMediaImageLoader;
import de.timfreiheit.mozart.model.MozartMediaProvider;
import de.timfreiheit.mozart.sample.ImageLoader;

public class MusicService extends MozartMusicService {

    private MediaNotificationManager mediaNotificationManager;

    @Override
    public MozartMediaProvider getMediaProvider() {
        return MediaProvider.getInstance();
    }

    @Override
    public MozartMediaNotificationManager getMediaNotificationManager() {
        if (mediaNotificationManager == null) {
            try {
                mediaNotificationManager = new MediaNotificationManager(this);
            } catch (RemoteException e) {
                e.printStackTrace();
            }
        }
        return mediaNotificationManager;
    }

    @Override
    public MozartMediaImageLoader getImageLoader() {
        return ImageLoader.getInstance();
    }
}

File: sample/src/main/java/de/timfreiheit/mozart/sample/player/MusicService.kt
package de.timfreiheit.mozart.sample.player

import de.timfreiheit.mozart.MozartMusicService
import de.timfreiheit.mozart.exoplayer.ExoplayerPlayback
import de.timfreiheit.mozart.model.MozartMediaProvider
import de.timfreiheit.mozart.model.image.MozartMediaImageLoader
import de.timfreiheit.mozart.playback.local.MediaPlayerPlayback
import de.timfreiheit.mozart.sample.ImageLoader

class MusicService : MozartMusicService() {

    override val mediaProvider: MozartMediaProvider
        get() = MediaProvider

    override val imageLoader: MozartMediaImageLoader
        get() = ImageLoader

    override fun createLocalPlayback() = ExoplayerPlayback(this)
}
